### PR TITLE
refactor: replace debounce function rooks debounce

### LINKED ISSUE #228: Refactor: Replace custom hooks with rooks
### Type of feature

🧑‍💻 Refactor

### Current behavior

We are using a custom hook for search, but @0-vortex called out the existence of this [package](https://react-hooks.org/docs/useDebounce).

https://github.com/open-sauced/hot/pull/215/files/a6d040095b2eed6863dd527596d00bebb9ceea4b#r928032778

### Suggested solution

1. Replace the [useDebounce](https://github.com/open-sauced/hot/blob/ui-v3-hero/src/hooks/useDebounce.ts) hook with [this](https://react-hooks.org/docs/useDebounce).
2. Leverage this library for this issue https://github.com/open-sauced/hot/issues/205

### Additional context

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Contributing Docs

- [X] I agree to follow this project's Contribution Docs

### PR DESCRIPTION
**Pull request by hokagedemehin**
## What type of PR is this? (check all applicable)

- [x] 🧑‍💻 Code Refactor

## Description

This PR refactors the debounce function with the rooks collection of hooks. I implemented useDebounce and useDidUpdate together to achieve the same thing as the current working solution

## Related Tickets & Documents
Fixes #228 

## Mobile & Desktop Screenshots/Recordings
debounce

## Added tests?
- [x] 🙅 no, because they aren't needed

## Added to documentation?
- [x] 🙅 no documentation is needed

## [optional] What gif best describes this PR or how it makes you feel?
spiderman

### CODE REVIEW

**Comment by bdougie** [comment]
How is it that this PR still touches 19 files @hokagedemehin? 

Also, this file should be deleted in favor of using rooks - src/hooks/useDebounce.ts

**Comment by hokagedemehin** [comment]
@bdougie when I ran this code `git pull upstream  <branch name>` it created a lot of files and also cause a merge conflict on some of the existing files I already have on my machine. I only added new code to the hero.tsx file and installed the necessary packages from rooks

You can check the changes to confirm my claims

File: index.html [hunk 1]
```diff
@@ -1,46 +1,71 @@
...
-    <link rel="apple-touch-icon" href="/apple-touch-icon.png" sizes="180x180"/>
-    <link rel="mask-icon" href="/favicon.svg" color="#FFFFFF"/>
-    <link rel="preconnect" href="https://fonts.googleapis.com">
-    <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
-    <link href="https://fonts.googleapis.com/css2?family=Inter:wght@100;200;300;400;500;600;700;800;900&display=swap" rel="stylesheet">
-    <meta name="msapplication-TileColor" content="#FFFFFF">
-    <meta name="theme-color" content="#FFFFFF" />
 
-    <meta property="dcterms:modified" content="<%= date %>" />
+<head>
```
Comment (NsdHSO) [inline index.html:4]: please you can remove lines: 20,21,22,23,... you have duplicate code. Please to did clean up.  Thanks.

**Comment by bdougie** [comment]
@0-vortex left an explanation on this in the discord. Because fixes are going to main, it is putting beta in a weird spot if you branch from main and PR to beta. 

This PR may need to wait for beta to be merged.

File: index.html (hunk 1 above)
Comment (hokagedemehin) [reply index.html:4]: I have removed the duplicate links in the index.html

**Comment by hokagedemehin** [comment]
Yes, I am talking with him on this now. 

Thanks

**Comment by 0-vortex** [comment]
the hero component reached the `beta` branch, will close this in favor of a rebase

**Pull request closed without merging**